payments/schema/mutations.py

```diff
@@ -413,7 +413,15 @@ def mutate_and_get_payload(cls, root, info, **input):
                 input["lease"] = lease
 
         try:
+            # handle case where input has both lease and status.
+            # set order status only after changing the lease, because setting order status
+            # usually triggers a change in lease status.
+            new_status = input.pop("status", None)
+            old_status = order.status
             update_object(order, input)
+            if new_status and new_status != old_status:
```

Comment:
The `set_status` already checks that the old and new statuses are different, so we can skip that check.